src/Akka.Hosting/Configuration/HoconConfigurationSection.cs

```diff
@@ -0,0 +1,83 @@
+﻿// -----------------------------------------------------------------------
+//   <copyright file="HoconConfigurationSection.cs" company="Petabridge, LLC">
+//     Copyright (C) 2015-2023 .NET Petabridge, LLC
+//   </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Primitives;
+
+namespace Akka.Hosting.Configuration;
+
+internal class HoconConfigurationSection: IConfigurationSection
```

Comment:
This class wraps the official `ConfigurationSection` class and automatically inspects the providers to see if they need to be converted to `EnvironmentVariablesConfigurationHoconAdapter`.